module/x/peggy/spec/04_messages.md

```diff
@@ -82,83 +65,59 @@ This message will fail if:
 - Failure to build a batch of transactions.
 - If the orchestrator address is not present in the validator set
 
-### MsgConfirmBatch
-
-When a `MsgRequestBatch` is observed, validators need to sign batch request to signify this is not a maliciously created batch and to avoid getting slashed. 
+### MsgSubmitConfirm
 
-+++ https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/msgs.proto#L137-143
-
-This message will fail if:
+When the peggy daemon witnesses an event on the counter chain the validator will submit confirmation of the event. 
 
-- The batch does not exist
-- If checkpoint generation fails
-- If a none validator address or delegated address 
-- If the counter chain address is empty or incorrect.
-- If counter chain address fails signature validation
-- If the signature was already presented in a previous message
+<!-- +++ https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/msgs.proto#L79-84 -->
 
-### MsgConfirmLogicCall
+```proto
+  message MsgSubmitClaim {
+    google.protobuf.Any confirm = 1 [
```

Comment:
wondering if this should be an array instead

Reply:
I can see this as being beneficial and potentially cost saving.

Reply:
+1 this should be an array. Orchestrators catching up after a period of downtime may need to submit hundreds of claims at a time.